tiny_dnn/core/framework/tensor.h

```diff
@@ -26,83 +25,67 @@ namespace tiny_dnn {
 
 /**
  * A tensor of the given dimension.
- * A tensor holds data in C-style nD array, i.e row-major order:
- * the rightmost index “varies the fastest”.
  *
- * Data is held by a std::vector with 64 bytes alignment.
- * Unmutable if kConst == true
+ * U is type of the data stored
+ * Storage is type of underlying storage
  */
-template <typename U         = float_t,
-          size_t kDimensions = 4,
-          bool kConst        = false,
-          typename Allocator = aligned_allocator<U, 64>>
+template <typename U = float_t, typename Storage = xt::xarray<U>>
 class Tensor {
-  // Define constant types for constant Tensor,
-  // and mutable ones for mutable Tensor
-  typedef typename std::conditional<kConst,
-                                    const TensorStorage<U, Allocator>,
-                                    TensorStorage<U, Allocator>>::type
-    TensorStorageType;
-  typedef typename std::conditional<kConst, const U *, U *>::type UPtr;
-  typedef typename std::shared_ptr<TensorStorageType> TensorStoragePointer;
-  typedef typename std::conditional<
-    kConst,
-    typename std::vector<U, Allocator>::const_iterator,
-    typename std::vector<U, Allocator>::iterator>::type StorageIterator;
+  typedef U *UPtr;
 
  public:
   /**
    * Initializes an empty tensor.
    * @return
    */
-  Tensor() {
-    offset_ = size_ = size_t(0);
-    storage_ptr_    = std::make_shared<TensorStorageType>();
-  }
+  Tensor() { storage_ = xt::xarray<U>(); }
```

Comment:
The decltype of `Tensor::storage_` is template parameter type `typename Storage = xt::xarray<U>`.
So assigning a specific type `xt::xarray<U>()` instance gives me a question.

But... it seems like `Tensor` class implmentors somehow want this library to be hardly dependent on xtensor library, so in that regard it may not be a big issue anymore. But then what's the reason for having the template parameter?

And I think using member initialization list is preferable.